### ISSUE TITLE
Equal Partition Card for Storage

### ISSUE THREAD

**Issue by shartte** [AppliedEnergistics/Applied-Energistics-2#5240]
Add an upgrade card for storage cells that makes it so the cell tries to equally partition it's storage across the partitioned items.

Use case: Make using a single drive for multiple bulk item types less of a shit-show. Avoid a scenario where a storage cell supposed to contain i.e. cobble and stone gets overflown with cobble making it not able to store stone.

**Comment by yueh** [comment]
I do not think we should add single purpose upgrade cards. They will just lead to more clutter. If we add more types, they should have multiple uses. Or at least have this intention.

Further what means trying? Will it balance how much of each type you can store. Say if they have stored 64 cobble, they will only store 64 -+ 8 dirt and to store more cobble you have to first add 72 dirt?

What I could see is adding something like a `Distribution Card`. Which goes into a couple different things. At first glance that would probably be the storage cells and export bus.
I never really liked tying the round robin mode to the capacity cards, but that was mostly because I did not want to add a new card just for it.

**Potential mechanics:**

## Export Bus

Would now be the required upgrade to enable round robin or other scheduling modes. Technically this would require someone to use at least 1 capacity card to make it worthwhile, but I could see changing the defaults to 3 slots, with the first capacity card providing +2 slots and +4 for the second.

Further I could see changing the modes to be a bit more useful. With a `Distribution Card` present this would include:

**Changing the slot limit from 1 to 64**
Allows for setting a ratio, e.g. export 8 cobble, then 1 coal

**Keep track of the exported amounts**
If the target only accepted 7 cobble, it will not continue with coal until it could export the missing cobble

**Take `Acceleration Card` into account**
If set to a 8:1 ratio and with one acceleration card, it should take this into account and export at most 8 items at at time, but limited to the currently processed type. So it will not roll over to export 1 coal and 7 cobble at once. Alternatively this could also affect the ratio, so with a single card it will not exprot 8 and then 1, but actually first 64 cobble (8*8) and then 8 coal before cycling again. This might make it a bit easier with the random mode.

**In combination with a `Redstone Card`**
Especially in pulse mode and in combination with an `Acceleration Card`, this could be used to export a set of different items with a know amount of pulses. Again with the 8:1 ration, two pulses with export everything it needs to smelt 8 cobble in a furnace. Assuming a single acceleration card or more. Otherwise it would still be 9 pulses. With something like 64:1 and 1 acceleration card it would mean 9 pulses, as each pulse with at most export 8 items.

## Import bus

Similar to the export bus, just mirrored for importing not exporting.

## Storage Cells

This would add a new mode on how the bytes are used. Currently I would see these modes

**Shared bytes for all types**

Basically the current behaviour, there would be a chance that it only stores cobble, even when set to other items

**Split bytes across types**

Bytes would be equally split across all configured types. So a 1k cell with 2 items configured would allocate 512 bytes for each type. Or about 16 with all slots configured (rounding down in this case)

*Definable ratio for each type (TBD)*

Similar to the export/import buses we could allow each slot to holde more than a stack size of one to define a ratio. So again 8 cobble, 1 coal would mean it allocates about 910 butes for cobble and 113 for coal

**Comment by mindforger** [comment]
I like the Distribution idea, especially the export one, this could probably make it into the interface too, so the blocking mode would become more intelligent on multislot devices that expose several slots to one side but require only a certain amount, and the item priority could match the slot number from left to right, Top to down in this order. This would invalidate a lot of „convenience“issues from the past
